examples/lock-app/efr32/include/LockManager.h

```diff
@@ -29,17 +29,73 @@
 
 #include <lib/core/CHIPError.h>
 
-using namespace ::chip;
+namespace EFR32DoorLock {
+namespace ResourceRanges {
+// Used to size arrays
+static constexpr uint16_t kMaxUsers                  = 10;
+static constexpr uint8_t kMaxCredentialsPerUser      = 10;
+static constexpr uint8_t kMaxWeekdaySchedulesPerUser = 10;
+static constexpr uint8_t kMaxYeardaySchedulesPerUser = 10;
+static constexpr uint8_t kMaxHolidaySchedules        = 10;
+static constexpr uint8_t kMaxCredentialSize          = 8;
+
+// Indices received for user/credential/schedules are 1-indexed
+static constexpr uint8_t kStartIndexValue = 1;
 
-// Currently up to 10 users are support on the EFR32 platform
-#define DOOR_LOCK_MAX_USERS 10
-#define DOOR_LOCK_MAX_CREDENTIAL_SIZE 8
-#define MININUM_USER_INDEX 1
-#define MINIMUM_CREDENTIAL_INDEX 1
-#define MAX_CREDENTIAL_PER_USER 10
-#define MAX_CREDENTIALS 50
+static constexpr uint8_t kMaxCredentials = kMaxUsers * kMaxCredentialsPerUser;
+} // namespace ResourceRanges
 
-static constexpr size_t DOOR_LOCK_CREDENTIAL_INFO_MAX_DATA_SIZE = 20;
+namespace LockInitParams {
+
+struct LockParam
+{
+    // Read from zap attributes
+    uint16_t numberOfUsers;
+    uint8_t numberOfCredentialsPerUser;
+    uint8_t numberOfWeekdaySchedulesPerUser;
+    uint8_t numberOfYeardaySchedulesPerUser;
+    uint8_t numberOfHolidaySchedules;
```

Comment:
init to 0 for all fields, to make sure our code paths are guaranteed deterministic.